topology/src/main/java/io/camunda/zeebe/topology/ClusterTopologyManagerService.java

```diff
@@ -43,6 +53,28 @@ public ClusterTopologyManagerService(
             config,
             clusterTopologyManager::onGossipReceived);
     clusterTopologyManager.setTopologyGossiper(clusterTopologyGossiper::updateClusterTopology);
+
+    isCoordinator = memberShipService.getLocalMember().id().id().equals(COORDINATOR_ID);
+  }
+
+  private TopologyInitializer getNonCoordinatorInitializer() {
+    return new FileInitializer(persistedClusterTopology)
+        .orThen(
+            new GossipInitializer(
+                persistedClusterTopology, clusterTopologyGossiper::updateClusterTopology));
+  }
+
+  private TopologyInitializer getCoordinatorInitializer(
+      final Supplier<Set<PartitionMetadata>> staticPartitionResolver,
+      final List<MemberId> knownMembers) {
+    return new FileInitializer(persistedClusterTopology)
+        .orThen(
+            new SyncInitializer(
+                persistedClusterTopology,
+                knownMembers,
+                this,
+                clusterTopologyGossiper::queryClusterTopology))
+        .orThen(new StaticInitializer(staticPartitionResolver, persistedClusterTopology));
```

Comment:
:thought_balloon: Do we not want/need a timeout on the `SyncInitializer`?

Reply:
No. It should get atleast one valid response from any member.